polyfill/lib/parse.mjs

```diff
@@ -0,0 +1,17 @@
+import { ES } from './ecmascript.mjs';
+const DateTime = ES.GetIntrinsic('%Temporal.DateTime%');
+const TimeZone = ES.GetIntrinsic('%Temporal.TimeZone%');
+
+export function parse(isoString) {
+  if (typeof isoString !== 'string') {
+    throw new TypeError('Expected string.');
+  }
+  const zIndex = isoString.indexOf('Z');
+  if (zIndex < 0) {
+    throw new RangeError('Not a valid ISO string.');
```

Comment:
I don't think this is correct. Strings without `Z` can be valid ISO strings, e.g. `2020-03-16T11:09:41.985169700-07:00` or `2020-03-16T11:09:41.985169700-07:00[America/Vancouver]` with our extra time zone name annotation.

Luckily, there's already a function which should be useful here: `ES.ParseIsoDateTime(isoString, { zoneRequired })`

Reply:
My bad, I misread the spec (over at IETF). Will fix.